gosub-bindings/include/gosub_api.h

```diff
@@ -18,9 +18,9 @@ struct render_tree_t {
   struct node_t *data;
 };
 
-/// Construct a new render tree.
-/// Returns NULL if unsuccessful.
-struct render_tree_t *render_tree_init(const char *html);
+/// Initialize a render tree by passing a stack-allocated
+/// struct by address.
+void render_tree_init(struct render_tree_t *render_tree, const char *html);
```

Comment:
I don't know too much about this, but shouldn't it be possible to create the `render_tree_t` in the function? I've seen this approach in many different C Libraries, so it's probably the best solution, when we want to have it on the stack.

Reply:
You can't return a pointer to something stack-allocated in a function, C doesn't transfer ownership. If you tried dereferencing it, you would segfault

Reply:
Unless they are doing something that I've never heard of before... I feel like that may be possible in C++ with its move semantics, but I haven't really seen anything like that in C

Reply:
Unless you mean returning the struct by value, I guess I didn't really think of that, but not something I do often since it seems kind of weird to copy the value around (even tho the compiler might do some optimizations on return values)

Reply:
Yes, after some thought, just returning the struct by value might be fine in this case, it's very small anyways

Reply:
ah, okay. I don't have much C experience. I mostly used C++ and not raw C.

Reply:
If you look at my personal github, I came from mostly raw C... but usually the structs I work with are relatively large so returning by value didn't come across my mind since I am so used to returning pointers

Reply:
> Yes, after some thought, just returning the struct by value might be fine in this case, it's very small anyways

I guess, you can just leave it the way it currently is, but if you want to change it just do it.

Reply:
I think I will leave it as a parameter and return the `int` to check if a failure occurred. Then I think this will be ready..